### PR TITLE
XW-3029 | featured video data returned by MercuryApi

### PR DESCRIPTION
**Pull request by rybmat**
https://wikia-inc.atlassian.net/browse/XW-3029

@Wikia/x-wing

### CODE REVIEW

**Review by hakubo — APPROVED** [review]

File: extensions/wikia/MercuryApi/MercuryApiController.class.php [hunk 1]
```diff
@@ -381,6 +381,12 @@ public function getPage() {
 				} else {
 					if ( !empty( $articleData['content'] ) ) {
 						$data['article'] = $articleData;
+						$data['article']['hasPortableInfobox'] = !empty( \Wikia::getProps( $title->getArticleID(), PortableInfoboxDataService::INFOBOXES_PROPERTY_NAME ) );
```
Comment (rogatty) [inline extensions/wikia/MercuryApi/MercuryApiController.class.php:384]: How about breaking this line so it's not so long?

**Review by rogatty — APPROVED** [review]

File: extensions/wikia/MercuryApi/MercuryApiController.class.php [hunk 2]
```diff
@@ -381,6 +381,12 @@ public function getPage() {
...
+
+						$featuredVideo = MercuryApiArticleHandler::getFeaturedVideoDetails( $title );
+						if ( !empty( $featuredVideo ) ) {
```
Comment (WikiaTech) [inline extensions/wikia/MercuryApi/MercuryApiController.class.php:387]: MAJOR Refactor this code to not nest more than 4 "if", "for", "while", "switch" and "try" statements. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS134)

File: extensions/wikia/MercuryApi/MercuryApiController.class.php [hunk 3]
```diff
@@ -381,6 +381,12 @@ public function getPage() {
...
+							$data['article']['featuredVideo'] = $featuredVideo;
+						}
 
 						if ( !$title->isContentPage() ) {
```
Comment (WikiaTech) [inline extensions/wikia/MercuryApi/MercuryApiController.class.php:391]: MAJOR Refactor this code to not nest more than 4 "if", "for", "while", "switch" and "try" statements. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS134)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 15 issues
* ![MAJOR][MAJOR] 6 major
* ![INFO][INFO] 9 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [MercuryApiController.class.php#L318](https://github.com/Wikia/app/blob/49d323bcbc2d7f00a109dc5db1d9444edeea8254/extensions/wikia/MercuryApi/MercuryApiController.class.php#L318): The Cyclomatic Complexity of this function "getPage" is 21 which is greater than 20 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1541)
1. ![MAJOR][MAJOR] [MercuryApiController.class.php#L331](https://github.com/Wikia/app/blob/49d323bcbc2d7f00a109dc5db1d9444edeea8254/extensions/wikia/MercuryApi/MercuryApiController.class.php#L331): Remove this commented out code. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS125)
1. ![MAJOR][MAJOR] [MercuryApiController.class.php#L400](https://github.com/Wikia/app/blob/49d323bcbc2d7f00a109dc5db1d9444edeea8254/extensions/wikia/MercuryApi/MercuryApiController.class.php#L400): Reduce this "switch/case" number of lines from 17 to at most 10, for example by extracting code into function. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1151)
1. ![MAJOR][MAJOR] [MercuryApiController.class.php#L412](https://github.com/Wikia/app/blob/49d323bcbc2d7f00a109dc5db1d9444edeea8254/extensions/wikia/MercuryApi/MercuryApiController.class.php#L412): Refactor this code to not nest more than 4 "if", "for", "while", "switch" and "try" statements. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS134)
1. ![INFO][INFO] [OoyalaConfigController.class.php#L220](https://github.com/Wikia/app/blob/49d323bcbc2d7f00a109dc5db1d9444edeea8254/extensions/wikia/ArticleVideo/OoyalaConfigController.class.php#L220): Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)
1. ![INFO][INFO] [OoyalaConfigController.class.php#L222](https://github.com/Wikia/app/blob/49d323bcbc2d7f00a109dc5db1d9444edeea8254/extensions/wikia/ArticleVideo/OoyalaConfigController.class.php#L222): Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)
1. ![INFO][INFO] [OoyalaConfigController.class.php#L224](https://github.com/Wikia/app/blob/49d323bcbc2d7f00a109dc5db1d9444edeea8254/extensions/wikia/ArticleVideo/OoyalaConfigController.class.php#L224): Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)
1. ![INFO][INFO] [OoyalaConfigController.class.php#L226](https://github.com/Wikia/app/blob/49d323bcbc2d7f00a109dc5db1d9444edeea8254/extensions/wikia/ArticleVideo/OoyalaConfigController.class.php#L226): Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)
1. ![INFO][INFO] [OoyalaConfigController.class.php#L228](https://github.com/Wikia/app/blob/49d323bcbc2d7f00a109dc5db1d9444edeea8254/extensions/wikia/ArticleVideo/OoyalaConfigController.class.php#L228): Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)
1. ![INFO][INFO] [OoyalaConfigController.class.php#L230](https://github.com/Wikia/app/blob/49d323bcbc2d7f00a109dc5db1d9444edeea8254/extensions/wikia/ArticleVideo/OoyalaConfigController.class.php#L230): Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)

[INFO]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO'
[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 15 issues
* ![MAJOR][MAJOR] 6 major
* ![INFO][INFO] 9 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [MercuryApiController.class.php#L318](https://github.com/Wikia/app/blob/71a988d664e313a66ade0e0fc69c9425189e268b/extensions/wikia/MercuryApi/MercuryApiController.class.php#L318): The Cyclomatic Complexity of this function "getPage" is 21 which is greater than 20 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1541)
1. ![MAJOR][MAJOR] [MercuryApiController.class.php#L331](https://github.com/Wikia/app/blob/71a988d664e313a66ade0e0fc69c9425189e268b/extensions/wikia/MercuryApi/MercuryApiController.class.php#L331): Remove this commented out code. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS125)
1. ![MAJOR][MAJOR] [MercuryApiController.class.php#L400](https://github.com/Wikia/app/blob/71a988d664e313a66ade0e0fc69c9425189e268b/extensions/wikia/MercuryApi/MercuryApiController.class.php#L400): Reduce this "switch/case" number of lines from 17 to at most 10, for example by extracting code into function. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1151)
1. ![MAJOR][MAJOR] [MercuryApiController.class.php#L412](https://github.com/Wikia/app/blob/71a988d664e313a66ade0e0fc69c9425189e268b/extensions/wikia/MercuryApi/MercuryApiController.class.php#L412): Refactor this code to not nest more than 4 "if", "for", "while", "switch" and "try" statements. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS134)
1. ![INFO][INFO] [OoyalaConfigController.class.php#L220](https://github.com/Wikia/app/blob/71a988d664e313a66ade0e0fc69c9425189e268b/extensions/wikia/ArticleVideo/OoyalaConfigController.class.php#L220): Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)
1. ![INFO][INFO] [OoyalaConfigController.class.php#L222](https://github.com/Wikia/app/blob/71a988d664e313a66ade0e0fc69c9425189e268b/extensions/wikia/ArticleVideo/OoyalaConfigController.class.php#L222): Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)
1. ![INFO][INFO] [OoyalaConfigController.class.php#L224](https://github.com/Wikia/app/blob/71a988d664e313a66ade0e0fc69c9425189e268b/extensions/wikia/ArticleVideo/OoyalaConfigController.class.php#L224): Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)
1. ![INFO][INFO] [OoyalaConfigController.class.php#L226](https://github.com/Wikia/app/blob/71a988d664e313a66ade0e0fc69c9425189e268b/extensions/wikia/ArticleVideo/OoyalaConfigController.class.php#L226): Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)
1. ![INFO][INFO] [OoyalaConfigController.class.php#L228](https://github.com/Wikia/app/blob/71a988d664e313a66ade0e0fc69c9425189e268b/extensions/wikia/ArticleVideo/OoyalaConfigController.class.php#L228): Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)
1. ![INFO][INFO] [OoyalaConfigController.class.php#L230](https://github.com/Wikia/app/blob/71a988d664e313a66ade0e0fc69c9425189e268b/extensions/wikia/ArticleVideo/OoyalaConfigController.class.php#L230): Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)

[INFO]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO'
[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 15 issues
* ![MAJOR][MAJOR] 6 major
* ![INFO][INFO] 9 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [MercuryApiController.class.php#L318](https://github.com/Wikia/app/blob/6fe6e7b3b7db56cffb0addb51113d2b9984c2c7d/extensions/wikia/MercuryApi/MercuryApiController.class.php#L318): The Cyclomatic Complexity of this function "getPage" is 21 which is greater than 20 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1541)
1. ![MAJOR][MAJOR] [MercuryApiController.class.php#L331](https://github.com/Wikia/app/blob/6fe6e7b3b7db56cffb0addb51113d2b9984c2c7d/extensions/wikia/MercuryApi/MercuryApiController.class.php#L331): Remove this commented out code. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS125)
1. ![MAJOR][MAJOR] [MercuryApiController.class.php#L400](https://github.com/Wikia/app/blob/6fe6e7b3b7db56cffb0addb51113d2b9984c2c7d/extensions/wikia/MercuryApi/MercuryApiController.class.php#L400): Reduce this "switch/case" number of lines from 17 to at most 10, for example by extracting code into function. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1151)
1. ![MAJOR][MAJOR] [MercuryApiController.class.php#L412](https://github.com/Wikia/app/blob/6fe6e7b3b7db56cffb0addb51113d2b9984c2c7d/extensions/wikia/MercuryApi/MercuryApiController.class.php#L412): Refactor this code to not nest more than 4 "if", "for", "while", "switch" and "try" statements. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS134)
1. ![INFO][INFO] [OoyalaConfigController.class.php#L220](https://github.com/Wikia/app/blob/6fe6e7b3b7db56cffb0addb51113d2b9984c2c7d/extensions/wikia/ArticleVideo/OoyalaConfigController.class.php#L220): Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)
1. ![INFO][INFO] [OoyalaConfigController.class.php#L222](https://github.com/Wikia/app/blob/6fe6e7b3b7db56cffb0addb51113d2b9984c2c7d/extensions/wikia/ArticleVideo/OoyalaConfigController.class.php#L222): Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)
1. ![INFO][INFO] [OoyalaConfigController.class.php#L224](https://github.com/Wikia/app/blob/6fe6e7b3b7db56cffb0addb51113d2b9984c2c7d/extensions/wikia/ArticleVideo/OoyalaConfigController.class.php#L224): Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)
1. ![INFO][INFO] [OoyalaConfigController.class.php#L226](https://github.com/Wikia/app/blob/6fe6e7b3b7db56cffb0addb51113d2b9984c2c7d/extensions/wikia/ArticleVideo/OoyalaConfigController.class.php#L226): Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)
1. ![INFO][INFO] [OoyalaConfigController.class.php#L228](https://github.com/Wikia/app/blob/6fe6e7b3b7db56cffb0addb51113d2b9984c2c7d/extensions/wikia/ArticleVideo/OoyalaConfigController.class.php#L228): Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)
1. ![INFO][INFO] [OoyalaConfigController.class.php#L230](https://github.com/Wikia/app/blob/6fe6e7b3b7db56cffb0addb51113d2b9984c2c7d/extensions/wikia/ArticleVideo/OoyalaConfigController.class.php#L230): Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)

[INFO]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO'
[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

File: extensions/wikia/MercuryApi/handlers/MercuryApiArticleHandler.class.php [hunk 4]
```diff
@@ -95,6 +95,31 @@ public static function getArticleJson( WikiaRequest $request, Article $article )
 		)->getData();
 	}
 
+	public static function getFeaturedVideoDetails( Title $article ): array {
```
Comment (kvas-damian) [inline extensions/wikia/MercuryApi/handlers/MercuryApiArticleHandler.class.php]: Please decide. Is a title or article ;)

File: extensions/wikia/MercuryApi/handlers/MercuryApiArticleHandler.class.php [hunk 5]
```diff
@@ -95,6 +95,31 @@ public static function getArticleJson( WikiaRequest $request, Article $article )
...
+		global $wgEnableArticleFeaturedVideo, $wgArticleVideoFeaturedVideos;
+
+		$title = $article->getPrefixedDBkey();
```
Comment (kvas-damian) [inline extensions/wikia/MercuryApi/handlers/MercuryApiArticleHandler.class.php]: This one definitely is not `title` ;)
Please rename those variables to make it less confusing.

**Comment by WikiaTech** [comment]
SonarQube analysis reported 15 issues
* ![MAJOR][MAJOR] 6 major
* ![INFO][INFO] 9 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [MercuryApiController.class.php#L318](https://github.com/Wikia/app/blob/8fbca2a295ef2ce2b477d376a29959e77fc67965/extensions/wikia/MercuryApi/MercuryApiController.class.php#L318): The Cyclomatic Complexity of this function "getPage" is 21 which is greater than 20 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1541)
1. ![MAJOR][MAJOR] [MercuryApiController.class.php#L331](https://github.com/Wikia/app/blob/8fbca2a295ef2ce2b477d376a29959e77fc67965/extensions/wikia/MercuryApi/MercuryApiController.class.php#L331): Remove this commented out code. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS125)
1. ![MAJOR][MAJOR] [MercuryApiController.class.php#L400](https://github.com/Wikia/app/blob/8fbca2a295ef2ce2b477d376a29959e77fc67965/extensions/wikia/MercuryApi/MercuryApiController.class.php#L400): Reduce this "switch/case" number of lines from 17 to at most 10, for example by extracting code into function. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1151)
1. ![MAJOR][MAJOR] [MercuryApiController.class.php#L412](https://github.com/Wikia/app/blob/8fbca2a295ef2ce2b477d376a29959e77fc67965/extensions/wikia/MercuryApi/MercuryApiController.class.php#L412): Refactor this code to not nest more than 4 "if", "for", "while", "switch" and "try" statements. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS134)
1. ![INFO][INFO] [OoyalaConfigController.class.php#L220](https://github.com/Wikia/app/blob/8fbca2a295ef2ce2b477d376a29959e77fc67965/extensions/wikia/ArticleVideo/OoyalaConfigController.class.php#L220): Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)
1. ![INFO][INFO] [OoyalaConfigController.class.php#L222](https://github.com/Wikia/app/blob/8fbca2a295ef2ce2b477d376a29959e77fc67965/extensions/wikia/ArticleVideo/OoyalaConfigController.class.php#L222): Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)
1. ![INFO][INFO] [OoyalaConfigController.class.php#L224](https://github.com/Wikia/app/blob/8fbca2a295ef2ce2b477d376a29959e77fc67965/extensions/wikia/ArticleVideo/OoyalaConfigController.class.php#L224): Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)
1. ![INFO][INFO] [OoyalaConfigController.class.php#L226](https://github.com/Wikia/app/blob/8fbca2a295ef2ce2b477d376a29959e77fc67965/extensions/wikia/ArticleVideo/OoyalaConfigController.class.php#L226): Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)
1. ![INFO][INFO] [OoyalaConfigController.class.php#L228](https://github.com/Wikia/app/blob/8fbca2a295ef2ce2b477d376a29959e77fc67965/extensions/wikia/ArticleVideo/OoyalaConfigController.class.php#L228): Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)
1. ![INFO][INFO] [OoyalaConfigController.class.php#L230](https://github.com/Wikia/app/blob/8fbca2a295ef2ce2b477d376a29959e77fc67965/extensions/wikia/ArticleVideo/OoyalaConfigController.class.php#L230): Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)

[INFO]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO'
[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

**Review by kvas-damian — APPROVED** [review]

**Pull request merged**